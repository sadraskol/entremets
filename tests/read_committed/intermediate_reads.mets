init do
    insert into users (id, age) values (1, 10)
end

process do
    begin read_committed
    update users set age := 101 where id = 1
latch
    commit
end

process do
    begin read_committed
    let age := select age from users where id = 1
    update users set age := 11 where id = 1
    commit
end

property = never[age = 101]
property = eventually<select age from users where id = 1 is 11>
