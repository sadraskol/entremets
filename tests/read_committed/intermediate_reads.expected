Following property was violated: eventually<select age from users where id = 1 is 11>